resources/js/table.js

```diff
@@ -0,0 +1,228 @@
+let tableHeads
+let tableContent
+/*------------------- Table Header loading fn -------------------*/
+
+let tableHeadLoader = function (headerList) {
+  if (headerList) {  // Checks whether a table exists.
+    tableHeads = headerList
+    let headRow = document.createElement("tr")
+
+    for (let headValue of headerList) {
+      let headElement = document.createElement("th")
+
+      // Adding text content and data attribute to 'th' element
+      headElement.textContent = headValue.tittle
+      headElement.setAttribute("data-type", headValue.type)
+      headElement.setAttribute("data-sortable", headValue.sortable)
+      headElement.id = headValue.id
+
+      // Adding up and down arrows
+      if (headValue.sortable == true) {
+
+        let upArrow = document.createElement('i')
+        upArrow.className = 'up Arrow'
+
+        let downArrow = document.createElement('i')
+        downArrow.className = 'down Arrow'
+
+        headElement.appendChild(upArrow)
+        headElement.appendChild(downArrow)
+      }
+
+      headRow.appendChild(headElement)
+    }
+
+    // Insert tr into thead
+    let tableHead = document.querySelector('thead')
+    tableHead.appendChild(headRow)
+
+    bindEventListener(tableHead)
+
+  }
+}
+
+/*------------------- Event listener for table headers loading fn -*/
+function bindEventListener(tableHead) {
+
+  tableHead.addEventListener("click", function (e) {
+    let header = e.target
```

Comment:
Bug: Click on sort-arrows are not triggering sort.

Add pointer cursor to sortable thead cells.

Reply:
Fixed